### PR TITLE
{bio}[foss/2016b] bcl2fastq2 2.19.1

### PR DESCRIPTION
**Pull request by SethosII**
This adds `bcl2fastq2` for the `foss-2016b` toolchain.

This is the strange software which needs `Boost` in version `1.54.0` (#4829) and has a nested archive (#4830).

### CODE REVIEW

File: easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+
+easyblock = 'ConfigureMake'
+
+name = 'bcl2fastq2'
+version = '2.19.1'
+versionsuffix = '-Python-2.7.12'
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb]: Please use the Python version macros i.e.
` '-Python-%(pyver)s'`

File: easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+homepage = 'https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html'
+description = """bcl2fastq Conversion Software both demultiplexes data and converts BCL files generated by
+ Illumina sequencing systems to standard FASTQ file formats for downstream analysis."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+source_urls = ['ftp://webdata2:webdata2@ussd-ftp.illumina.com/downloads/software/bcl2fastq/']
+sources = [{
+    'filename': '%(name)s-v%(version)s-tar.zip',
+    'extract_cmd': 'unzip -p %s | tar -xzvf -', # source file is a .zip that contains a .tar.gz
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb]: Should be two spaces before the comment here

File: easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}]
+
+checksums = ['cf13580f2c1ebcc3642b4d98a02ad01e41a44e644db7d31730f9767b25521806']
+
+start_dir = 'src'
+configopts = '--force-builddir'
+
+dependencies = [
+    ('Boost', '1.54.0', versionsuffix),
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.19.1-foss-2016b-Python-2.7.12.eb]: Add `Python` dependency here please

**Comment by SethosII** [comment]
@verdurin I added the requested changes.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
ca014.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4f3b4ceece85553c94267f95cad5aae6 for a full test report.

**Comment by verdurin** [comment]
Looks good

**Comment by verdurin** [comment]
Going in, thanks @SethosII

**Pull request merged**